From 7e5d190ccce8dc064d5756225e306e65fa534ae9 Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Mon, 2 Apr 2012 08:20:34 -0400
Subject: [PATCH] check for proper return from dirent_ensure_type

Fixes 'systemctl list-unit-files', which previously returned only:

  Failed to issue method call: No such file or directory
---
 src/install.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/install.c b/src/install.c
index 174d79b..9256116 100644
--- a/src/install.c
+++ b/src/install.c
@@ -1853,7 +1853,7 @@ int unit_file_get_list(
 
                         r = dirent_ensure_type(d, de);
                         if (r < 0) {
-                                if (errno == ENOENT)
+                                if (r == -ENOENT)
                                         continue;
 
                                 goto finish;
-- 
1.7.9.5

